app/containers/Settings/Settings.jsx

```diff
@@ -102,28 +105,33 @@ export default class Settings extends Component<Props, State> {
     setCurrency(e.target.value)
   }
 
-  deleteWallet = (key: string) => {
-    const { setKeys, showModal } = this.props
+  deleteWalletAccount = (label: string, key: string) => {
+    const { setAccounts, showModal } = this.props
     showModal(MODAL_TYPES.CONFIRM, {
       title: 'Confirm Delete',
-      text: `Please confirm deleting saved wallet - ${key}`,
+      text: `Please confirm deleting saved wallet - ${label}`,
       onClick: () => {
         // eslint-disable-next-line
-        storage.get('keys', (error, data) => {
-          delete data[key]
-          storage.set('keys', data)
-          setKeys(data)
+        storage.get('userWallet', (error, data) => {
+          data.accounts.map((account, index) => {
```

Comment:
Can we do some error handling here?
` if (error) { ... } else { delete wallet account }`
Also can we use something like:
`const keys = reject(data.accounts, { key });`
Also, why do we need to check the label when deleting? isn't a key is unique?